scr/main/java/bittercode/util/StoreUtil.java

```diff
@@ -65,7 +65,7 @@ else if (!items.contains(selectedBookId))
                 int itemQty = 0;
 if (session.getAttribute("qty_" + selectedBookId) != null)
                     itemQty = (int) session.getAttribute("qty_" + selectedBookId);
-                if (itemQty > 1) {
+                if (itemQty == 1) {
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Cart removal logic may be incorrect</b></div>
<div id="fix">

The logic for removing items from cart seems inverted. When `itemQty == 1`, the item should be completely removed rather than decremented
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">

```
 @@ -68 +68 @@
 -                if (itemQty == 1) {
 +                if (itemQty > 1) {
```


```suggestion
                if (itemQty > 1) {
```



</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/9#issuecomment-2527843320>#0f79b3</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged